File: instrumentation/okhttp/okhttp-3.0/library/src/main/java/io/opentelemetry/instrumentation/library/okhttp/v3_0/OkHttpWebsocketInstrumentation.java
/*
 * Copyright The OpenTelemetry Authors
 * SPDX-License-Identifier: Apache-2.0
 */

package io.opentelemetry.instrumentation.library.okhttp.v3_0;

import androidx.annotation.NonNull;
import com.google.auto.service.AutoService;
import io.opentelemetry.android.instrumentation.AndroidInstrumentation;
import io.opentelemetry.android.instrumentation.InstallationContext;
import io.opentelemetry.instrumentation.library.okhttp.v3_0.internal.WebsocketEventGenerator;

@AutoService(AndroidInstrumentation.class)
public class OkHttpWebsocketInstrumentation implements AndroidInstrumentation {
    @Override
    public void install(@NonNull InstallationContext ctx) {
        WebsocketEventGenerator.configure(ctx);
    }
}

File: instrumentation/okhttp3-websocket/library/src/main/java/io/opentelemetry/instrumentation/library/okhttp/v3_0/websocket/internal/OkHttpWebsocketInstrumentation.kt
/*
 * Copyright The OpenTelemetry Authors
 * SPDX-License-Identifier: Apache-2.0
 */

@file:Suppress("ktlint:standard:package-name")

package io.opentelemetry.instrumentation.library.okhttp.v3_0.websocket.internal

import com.google.auto.service.AutoService
import io.opentelemetry.android.instrumentation.AndroidInstrumentation
import io.opentelemetry.android.instrumentation.InstallationContext

@AutoService(AndroidInstrumentation::class)
class OkHttpWebsocketInstrumentation : AndroidInstrumentation {
    override fun install(ctx: InstallationContext) {
        WebsocketEventGenerator.configure(ctx)
    }

    override val name: String = "okhttp-websocket"
}
